### ISSUE TITLE
Not working with Node.js 13.5.0

### ISSUE THREAD

**Issue by mohd-akram** [nexe/nexe#712]
**What happened**:
nexe fails to build.

```
C:\Users\user\.nexe\13.5.0\src\histogram-inl.h(46,33): warning C4244: 'return': conversion from 'int 64_t' to 'double', possible loss of data [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
C:\Users\user\.nexe\13.5.0\src\histogram-inl.h(54,18): warning C4244: 'initializing': conversion fro m 'int64_t' to 'double', possible loss of data [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
C:\Users\user\.nexe\13.5.0\src\node_revert.h(37,3): warning C4065: switch statement contains 'defaul t' but no 'case' labels [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
C:\Users\user\.nexe\13.5.0\src\node.cc(868,66): error C2059: syntax error: ')' [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
C:\Users\user\.nexe\13.5.0\src\node.cc(876,67): error C2059: syntax error: ')' [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
C:\Users\user\.nexe\13.5.0\src\node.cc(1028,48): warning C4244: 'argument': conversion from 'int64_t ' to 'int', possible loss of data [C:\Users\user\.nexe\13.5.0\libnode.vcxproj]
```

What `node.cc` looks like:

```c++
    const int exit_code = 0;//ProcessGlobalArgs(&env_argv,
                                            nullptr,
                                            errors,
                                            kAllowedInEnvironment);
```

**What you expected to happen**:
nexe works.

**How to reproduce it (as minimally and precisely as possible)**:
On Windows:

    echo console.log('hello world'); > index.js && nexe --build --verbose index.js

**Anything else we need to know?**:
The `ProcessGlobalArgs` changes in `node.cc` seem to have been introduced in 13.2.0.

**Environment**:
- Platform(OS/Version): Windows 10 Version 1903 (OS Build 18362.535)
- Host Node Version: 13.5.0
- Target Node Version: 13.5.0
- Nexe version: 3.3.2
- Python Version: 3.8.1

**Comment by gautaz** [comment]
The same error occurs for linux builds for NodeJS 13.5.0, 13.6.0 and 13.7.0 (I did not check with older releases):

```
../src/node.cc:898:66: error: expected ';' before ')' token
                                             kAllowedInEnvironment);
                                                                  ^
```

The build environment used to reproduce this can be operated with Docker, here is the Dockerfile:

```
FROM debian:buster-20200130-slim

ARG NODE_JS_VERSION=13.7.0
ARG NEXE_TARGET=linux-x64-${NODE_JS_VERSION}

ENV DEBIAN_FRONTEND=noninteractive
ENV APT_KEY_DONT_WARN_ON_DANGEROUS_USAGE=DontWarn

SHELL ["/bin/bash", "-o", "pipefail", "-c"]

RUN apt-get update
RUN apt-get install --no-install-recommends -y build-essential ca-certificates curl linux-headers-amd64 python
RUN curl -sL https://deb.nodesource.com/setup_13.x | bash -
RUN apt-get install --no-install-recommends -y nodejs=${NODE_JS_VERSION}-1nodesource1
RUN npm install -g nexe@3.3.2
RUN touch index.js
RUN nexe --build --verbose --target ${NEXE_TARGET} --input index.js --output output --make="-j$(($(grep -c "processor" /proc/cpuinfo) + 1))"
```

**Comment by calebboyd** [comment]
This is typical of the current release line, which is only active for 6 months. It will be fixed for the next LTS release (node 14).

https://nodejs.org/en/about/releases/

**Comment by gautaz** [comment]
@calebboyd You are right, I totally understand that this should not be necessarily fixed by the build process of `nexe` and I agree that this issue should be raised in the NodeJS project if it can be reproduced without the `nexe` build system.

In the meantime, if you have any idea of which options might be used to work around the problem, that would be of great help. It would probably help part of us who are preparing for the next NodeJS release...

**Comment by calebboyd** [comment]
Its definitely an issue with the nexe patch, you should be able to avoid it by passing `--enableNodeCli` assuming your application isn't executed with its own CLI options.

**Comment by gautaz** [comment]
Checked with `--enableNodeCli`, it works :smile:.
@calebboyd Thanks a lot for the tip.
Still need to cope with the CLI options issue but it can be easily worked around for test purposes.

**Comment by vladiks** [comment]
I am facing exact same problem when building node 12.16.1 on ubuntu on arch64. Nexe version 3.3.2.

**Comment by ricardopolo** [comment]
@calebboyd with the new release of node 14, are you planning to fix it for this version that will be LTS?

**Comment by calebboyd** [comment]
When I get time, Hopefully before October (LTS). PRs welcomed

**Comment by ricardopolo** [comment]
Thanks @calebboyd. I would love to do a PR but sadly I am no too good with c++.

**Comment by mohd-akram** [comment]
Fixed in #742.
